pkg/volume/csi/csi_block.go

```diff
@@ -252,21 +252,70 @@ func (m *csiBlockMapper) SetUpDevice() (string, error) {
 	}
 
 	// Call NodeStageVolume
-	stagingPath, err := m.stageVolumeForBlock(ctx, csiClient, accessMode, csiSource, attachment)
+	_, err = m.stageVolumeForBlock(ctx, csiClient, accessMode, csiSource, attachment)
 	if err != nil {
 		return "", err
 	}
 
+	return m.getPublishPath(), nil
+}
+
+func (m *csiBlockMapper) MapDevice(devicePath, globalMapPath, volumeMapPath, volumeMapName string, podUID types.UID) error {
+	// Get csiSource from spec
+	if m.spec == nil {
+		klog.Error(log("blockMapper.MapDevice spec is nil"))
+		return fmt.Errorf("spec is nil")
+	}
+
+	csiSource, err := getCSISourceFromSpec(m.spec)
+	if err != nil {
+		klog.Error(log("blockMapper.MapDevice failed to get CSI persistent source: %v", err))
+		return err
+	}
+
+	driverName := csiSource.Driver
+	skip, err := m.plugin.skipAttach(driverName)
+	if err != nil {
+		return errors.New(log("blockMapper.MapDevice failed to check CSIDriver for %s: %v", driverName, err))
+	}
+
+	// Search for attachment by VolumeAttachment.Spec.Source.PersistentVolumeName
+	var attachment *storage.VolumeAttachment
+	if !skip {
+		// Search for attachment by VolumeAttachment.Spec.Source.PersistentVolumeName
+		nodeName := string(m.plugin.host.GetNodeName())
+		attachID := getAttachmentName(csiSource.VolumeHandle, csiSource.Driver, nodeName)
+		attachment, err = m.k8s.StorageV1().VolumeAttachments().Get(attachID, meta.GetOptions{})
+		if err != nil {
+			return errors.New(log("blockMapper.MapDevice failed to get volume attachment [id=%v]: %v", attachID, err))
+		}
+	}
+
+	//TODO (vladimirvivien) implement better AccessModes mapping between k8s and CSI
+	accessMode := v1.ReadWriteOnce
+	if m.spec.PersistentVolume.Spec.AccessModes != nil {
+		accessMode = m.spec.PersistentVolume.Spec.AccessModes[0]
```

Comment:
Do we care about other elements in AccessModes (len(m.spec.PersistentVolume.Spec.AccessModes) > 1) ?

Reply:
I agree that this should be imporved. 

But I think that this needs to be updated for both filesystem and block, in the same way.
(https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/csi/csi_mounter.go#L176)

So, this would be outside the scope of this PR.
Let's improve this in a separate PR.